vmm/src/igvm/igvm_loader.rs

```diff
@@ -64,6 +71,41 @@ enum ParameterAreaState {
     Inserted,
 }
 
+#[cfg(feature = "sev_snp")]
+fn igvm_memmap_from_ram_range(ram_range: (u64, u64)) -> IGVM_VHS_MEMORY_MAP_ENTRY {
+    assert!(ram_range.0 % HV_PAGE_SIZE == 0);
+    assert!((ram_range.1 - ram_range.0) % HV_PAGE_SIZE == 0);
+
+    IGVM_VHS_MEMORY_MAP_ENTRY {
+        starting_gpa_page_number: ram_range.0 / HV_PAGE_SIZE,
+        number_of_pages: (ram_range.1 - ram_range.0) / HV_PAGE_SIZE,
+        entry_type: MemoryMapEntryType::MEMORY,
+        flags: 0,
+        reserved: 0,
+    }
+}
+
+#[cfg(all(feature = "sev_snp", target_arch = "x86_64"))]
```

Comment:
Good reminder about checking on the `target_arch` when `sev_snp` is enabled. I think we should add a global compile-time check to avoid compile `igvm` on `aarch64`, like what we did for avoiding compiling `tdx` with `sev_snp`. As I am assuming you are planning to support feature `igvm` (and hence `sev_snp`) only for `x86_64`, right?

Reply:
`igvm` is architecture independent but true that `sev_snp` is strictly `x86_64`. I have added a compile time for the same.

Reply:
Then you don't need to explicitly include `target_arch = "x86_64"` here for the feature guard.